### PR TITLE
Nneo wrangle fix

### LINKED ISSUE #4: use fauxpas for errors

### PR DESCRIPTION
**Pull request by jaroberti**
## Description

Updates to nneo_wrangle function.  Sensor-based measurement streams have more than 1 replicate measurement were being overwritten and the merge was adding extra rows to the tibble rather than merging by time stamp.  Also, there was an issue with the time filtering at the end.  This has been fixed 

## Related Issue

@sckott, These fixes should mitigate the issues in #10 for nneo_wrangle.

## Example

#the following has been added to the testthat script and works as planned.
nneo::nneo_wrangle(site_code="STER",time_start="2017-03-04", data_var="temperature",time_agr=30)

### CODE REVIEW

**Comment by sckott** [comment]
👍 thanks @jaroberti 

will have a look asap

File: R/nneo_wrangle.R [hunk 1]
```diff
@@ -108,20 +111,57 @@ nneo_wrangle<-function(site_code="BART",time_start="2016-06-20",time_end=NULL,
...
+    names(data_all)<-spatial_terms
+    #append spatial terms to col names:
+    for(i in 1:length(data_all)){
+            names(data_all[[i]])<-c(names(data_all[[i]][,1:2]),
+                                    paste0(names(data_all[[i]][3:length(data_all[[i]])]),".",spatial_terms[i]))
+            #set row.names to NULL:
+            row.names(data_all[[i]])<-NULL
+    }
+
+    # #create 'true' time sequence to merge to (in case of missing time stamps)
```
Comment (sckott) [inline R/nneo_wrangle.R]: @jaroberti can we remove all this commented out code starting here? or do you want to save for something?

**Comment by sckott** [comment]
@jaroberti seems like the output column names have changed. e.g. 

```r
'data.frame':	1537 obs. of  322 variables:
 $ startDateTime             : chr  "2016-06-20T00:00:00Z" "2016-06-20T00:30:00Z" "2016-06-20T01:00:00Z" "2016-06-20T01:30:00Z" ...
 $ endDateTime               : chr  "2016-06-20T00:30:00Z" "2016-06-20T01:00:00Z" "2016-06-20T01:30:00Z" "2016-06-20T02:00:00Z" ...
 $ difRadMean.000.060.x      : num  14.9 3.5 1.4 2.2 0.9 1.4 1.4 0.4 1.1 1.5 ...
 $ difRadMinimum.000.060.x   : num  3 0 0 0 0 0 0 0 0 0 ...
 $ difRadMaximum.000.060.x   : num  27.4 7.3 4.6 4.6 3 4.5 4.5 3 4.6 4.6 ...
```

is that intended? to have names like `difRadMaximum.000.060.x`. If so, what does that syntax mean? x is for longitude? and y for latitude?

UPDATE: the man file does have some info about the `000.060`, so that seems all good, but what does x and y refer to?  can that be added to the `@return` block?

**Comment by sckott** [comment]
wrt tibbles vs. data.frames: you've switched back to regular data.frame in this PR. the other fxns all returns tibble's now. Do you have a stong pref for one or the other?  If not, I think we should stick with tibbles so that outputs are consistent.

**Comment by jaroberti** [comment]
@sckott, that's very odd about the .x at the end of the DP names.  I'll have to look into that.  They're supposed to be in the format of [###].[###].  These numbers represent the horizontal and vertical identifiers of where the sensors are located on the tower.  This is important because many of our measurement-types, (e.g., air temperature, wind, PAR, etc.) are sampled at multiple locations within a site.  So, for example, the 000.060 in the radiation example above indicates that the radiation measurement was made at the tower (i.e., 000.) on the sixth measurement level (i.e., .060).  For a radiation measurement made by a sensor at ground level in the 3rd soil plot, for example, we'd see something along the lines of 003.000 (i.e., 003 = 3rd soil plot; 000 = ground level).  

I'm trying to figure out why the .x and .y are there.  I'll take a look at that.

My apologies, I meant to have these output as tibbles, not DFs.  I'll make the switch back to Tibbles.  I'll try to have the updates done within  the next 48 hours!

**Comment by sckott** [comment]
> They're supposed to be in the format of [###].[###]. These numbers represent the horizontal and vertical identifiers of where the sensors are located on the tower.

Okay, makes sense. But one thought is that those two pieces of information (tower number, mesaurement level on tower) seem like data, or at least can be considered metadata. Thus, they could be columns (e.g, `tower_number`, `tower_level`), and the values are values in the rows under those columns, e..g, 

```r
  tower_number tower_level ... (other columns)
       000.060           1                ...
       000.060           2                ...
       000.060           3                ...
```

or does that not make sense?

**Comment by jaroberti** [comment]
@sckott Yes, that information is definitely metadata.  I was trying to avoid something like that because I'm merging the data by time stamp.  If I add columns like that then I can't merge multiple spatial locations into a data frame structure (to my knowledge).

**Comment by jaroberti** [comment]
@sckott.  I've found a solution to my issue and will submit a pull request.

**Comment by sckott** [comment]
> If I add columns like that then I can't merge multiple spatial locations into a data frame structure 

where is there spatial information in the output data.frame?

**Comment by jaroberti** [comment]
@sckott I've kept that information in the column headers.

**Comment by sckott** [comment]
okay, loooks good

**Comment by jaroberti** [comment]
@sckott Thanks for merging everything.  One thing I'd like to dbl check with you is whether or not you think the output from nneo_wrangle is too clunky.  I output the data in tibble framework (as opposed to a list), to follow your framework, however, I can easily back track and output individual tibbles nested within a list.  Each nested tibble would then be a complete time series of an individual measurement-type.  This would mitigate the need for all columns to have the ###.### suffixes.  I could move these to the names of the nested tibbles rather than individual columns.  Let me know your thoughts.

**Comment by sckott** [comment]
i like the idea of the list of tibbles - if you like that idea better too, maybe implement that?

**Comment by jaroberti** [comment]
@sckott sounds good.  Might be a few days to a week before I'll have time to make the changes but I'll keep you posted.

**Pull request merged**